### DISCUSSION TITLE
Why 50 threads by default?

### DISCUSSION THREAD

**Discussion by videoman** [epi052/feroxbuster#311] [General]
Using 50 threads seems rather high as a default. Wouldn't it be better to have a default of like 10 or 15 threads as a default?

**Comment by epi052** [comment]
As far as resource usage, since ferox uses async/await, they're not full-blown OS threads (more can be read [here](https://github.com/epi052/feroxbuster#threads-and-connection-limits-at-a-high-level)). 

From a user perspective, you can easily set the default you prefer via [ferox-config.toml](https://github.com/epi052/feroxbuster#ferox-configtoml).  You can set the default at a number of 'levels':

- everyone shares the config
- per-user config
- per-target config 

If you wanted, you could tone it down for Target-A and have it cranked up for Target-B, etc... 

The actual reasoning behind why I chose 50? 🤷 It's roughly in line with similar tools and some value needed to be chosen. 

Is 10 or 15 a better option? Perhaps. But we could easily be having the same conversation about 10 or 15. 

An arbitrary default needed to be there, and it was designed to be very configurable in the event that the default doesn't suit a user's preferences.

**Comment by videoman** [comment]
I would highly recommend moving this to a much lower default... 50 threads could potentially take out a web server that maybe doesn't have the resources pretty quickly. In all of my testing, I try limit stuff to 10-15 threads by default, and then ramp up if the system looks to be stable.
